src/vsr/replica.zig

```diff
@@ -5013,35 +5014,41 @@ pub fn ReplicaType(
             return vsr.Checkpoint.trigger_for_checkpoint(self.op_checkpoint_next()).?;
         }
 
+        /// Returns the highest op that this replica can safely prepare to its WAL.
+        fn op_checkpoint_next_border(self: *const Self) u64 {
```

Comment:
Now that we dont' have "border prepares", the "border" name doesn't seem really great, though I can't suggest a better alternative...

Maybe:

- trigger --- the `op` when the checkpoints starts
- due --- the `op` by which the checkpoint should have completed

Or maybe even _not_ introduce any name here? 


```
self.op_checkpoint_next_trigger() + constants.pipeline_prepare_queue_max
```

actually reads great to me: trigger, plus whatever we can pipeline over the trigger.

Reply:
In the next PR in the sequence (which actually enables "prepare beyond the checkpoint") `op_checkpoint_next_border` is used _very_ frequently, so imo having it in a helper function rather than inlining `self.op_checkpoint_next_trigger() + constants.pipeline_prepare_queue_max` is quite useful.

It is used in ways that "trigger" is often used right now -- "the highest op that we will accept in our WAL". So I don't think "trigger" or "due" make sense. Maybe "limit" or "ceiling"?

Reply:
:thinking: yeah, `due` has inverted semantics here -- checkpoint is indeed due to be done by the time we accept that `op` into our WAL, _but_, in code, we don't block until checkpoint is ready, we  rather drop prepares. 

Can't say that `limit` or `ceiling` is much better than the border. Hm, maybe

```
op_checkpoint_pipeline_limit
```

? I think "pipeline" is key here. Maybe even just 


```
op_pipeline_limit
```

? The fact that it depends on the checkpoint is sort of an implementation detail I think?

Though, no strong opinion either way, we can live with `border_prepare` as well .

Reply:
:+1: I'm going to leave it as "border" at least for now. The fact that it is a term that is not used elsewhere in the code (so can't be confused with anything else) makes up for the fact that it is not self-explaining.